pkg/apiserver/handlers.go

```diff
@@ -281,17 +307,40 @@ func (tw *baseTimeoutWriter) timeout(msg string) {
 			enc := json.NewEncoder(tw.w)
 			enc.Encode(errors.NewServerTimeout(api.Resource(""), "", 0))
 		}
+	} else {
+		// The timeout writer has been used by the inner handler. There is
+		// no way to timeout the HTTP request at the point. We have to shutdown
+		// the connection for HTTP1 or reset stream for HTTP2.
+		//
+		// Note from: Brad Fitzpatrick
```

Comment:
panic in ServeHTTP. We want to throw that out directly to go HTTP Server, so it can gracefully kill the connection/stream. Right now we catch all panics in the handler, so it wont propagate to HTTP Server at all.